src/neighborhood_search.jl

```diff
@@ -181,14 +182,20 @@ function foreach_point_neighbor(f::T, system_coords, neighbor_coords, neighborho
     @boundscheck checkbounds(system_coords, ndims(neighborhood_search), points)
 
     @threaded parallelization_backend for point in points
-        # Now we can safely assume that `point` is inbounds
-        @inbounds foreach_neighbor(f, system_coords, neighbor_coords,
-                                   neighborhood_search, point)
+        # This check is optimized away when `points_active` is `nothing`
+        if point_active(points_active, point)
+            # After the explicit boundscheck, we can safely assume that `point` is inbounds
+            @inbounds foreach_neighbor(f, system_coords, neighbor_coords,
+                                    neighborhood_search, point)
+        end
     end
 
     return nothing
 end
 
+@inline point_active(::Nothing, _) = true
+@inline point_active(points_active, point) = points_active[point]
```

Comment:
```suggestion
@inline point_active(points_active, point) = points_active[point] == true
```
In case we use UInt32 instead of Bool. (CUDA doesn't support atomic operations on bool)